### PR TITLE
Use default parameters where possible and other microrefactoring

### PR DESCRIPTION
**Pull request by thorn0**
## Description

Just refactoring.

## Checklist

- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
[`prefer-default-parameters`](
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-default-parameters.md)
, this rule maybe not very safe , but  can help .

File: src/language-markdown/printer-markdown.js [hunk 1]
```diff
@@ -503,9 +503,7 @@ function getNthListSiblingIndex(node, parentNode) {
   );
 }
 
-function getNthSiblingIndex(node, parentNode, condition) {
-  condition = condition || (() => true);
-
+function getNthSiblingIndex(node, parentNode, condition = (_) => true) {
```
Comment (fisker) [inline src/language-markdown/printer-markdown.js:506]: This default value not needed. Only used in this file, and it passed the argument.

File: src/language-markdown/utils.js [hunk 2]
```diff
@@ -214,7 +212,7 @@ function mapAst(ast, handler) {
     }
 
     return newNode;
-  })(ast, null, null);
+  })(ast, null);
```
Comment (fisker) [inline src/language-markdown/utils.js:215]: Let's pass an empty array?

File: tests_integration/runPrettier.js [hunk 3]
```diff
@@ -115,9 +115,7 @@ function runPrettier(dir, args = [], options = {}) {
 
   const result = { status, stdout, stderr, write };
 
-  const testResult = (testOptions) => {
-    testOptions = testOptions || {};
-
+  const testResult = (testOptions = {}) => {
```
Comment (fisker) [inline tests_integration/runPrettier.js:118]: Can this be removed? I didn't see any test without it.

**Review by fisker — APPROVED** [review]
Left few comments.

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**